### DISCUSSION TITLE
NNUE/HCE flag

### DISCUSSION THREAD

**Discussion by AlexBrunetti** [PGG106/Alexandria#480] [Q&A] [answered]
Hi, is there a way to switch from NNUE to HCE in versions 2.1 to 3.01 without typing 'nnue' in the console or recompiling the source code?

Thanks,
Alex

**Comment by PGG106** [comment]
I haven't looked at that code in a while, iirc no, it wasn't meant to be a user facing feature since there was (at least according to my judgment at the time) no good reason to actually use the hce code.
If you really need it I can expose it as a uci option and add that in the release.

**Reply by PGG106** [reply]
On a related note, you can find the same hce (which was a very basic pesto ripoff) under the branch "pestalex", which is based on a much more recent, stronger and stable version of Alex

**Reply by AlexBrunetti** [reply]
Hi, yes, I really need it because I exclusively test HCE engines for my list, either pure or with NN deactivatable, like in this case. There are 8 versions for which the UCI option would be convenient, but please know there's no urgency!

Thanks,
Alex

**Reply by PGG106** [reply]
To be quite honest I don't really recommend using any Alexandria version < 2.4, several stability issues took a while to properly fix,  give me a couple of days and I'll tweak the versions that are stable enough.

**Reply by PGG106** [reply, marked as answer]
I have added a new release for 3.0.1 that exposes nnue as a toggleable option <https://github.com/PGG106/Alexandria/releases/tag/v3.0.1> , note that as for any version before 3.0.2 the movegen has a minor bug that can lead to crashes.

**Reply by PGG106** [reply]
As for even older versions i don't think the work to update them is worth it, the hce was very unoriginal, the search rudimentary and the codebase quite terrible.

**Reply by AlexBrunetti** [reply]
Thank you very much for your effort.

Alex
